### ISSUE TITLE
PochiTrainer / PochiPredictor の責務分割

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#129]
## 概要

PR #127 で CheckpointStore / MetricsTracker を分離, PR #128 で PochiPredictor の継承を解消した.
しかし `PochiTrainer` (842行) と `PochiPredictor` (428行) にはまだ本来の責務外のロジックが残っており, 神クラス化の兆候がある. 各クラスを理想形 (`train()` + `train_epoch()` / `predict()` のみ) に近づけたい.

### PochiTrainer (842行) に残る責務外メソッド

| 分類 | メソッド | 行数 |
|------|---------|------|
| 検証 | `validate()`, `_compute_confusion_matrix_pytorch()`, `_log_confusion_matrix()` | ~100行 |
| セットアップ | `setup_training()`, `_build_layer_wise_param_groups()`, `_get_layer_group()`, `_log_layer_wise_lr()`, `_get_base_learning_rate()`, `is_layer_wise_lr_enabled()`, `get_layer_wise_learning_rates()` | ~250行 |
| CheckpointStore ラッパー | `save_checkpoint()`, `save_best_model()`, `save_last_model()`, `load_checkpoint()` | ~50行 |
| ワークスペース操作 | `save_training_config()`, `save_dataset_paths()`, `save_image_list()`, `get_workspace_info()` | ~80行 |

### PochiPredictor (428行) に残る責務外メソッド

| 分類 | メソッド | 行数 |
|------|---------|------|
| 結果出力 | `export_results_to_workspace()`, `save_confusion_matrix_image()` | ~100行 |
| データ準備+推論 | `predict_with_paths()` | ~50行 |
| 精度計算 | `calculate_accuracy()` | ~25行 |
| ワークスペース操作 | `_ensure_inference_workspace()`, `get_inference_workspace_info()` | ~25行 |

### Phase 1: Evaluator の抽出

`training/evaluator.py` を新規作成し, 検証ロジックを分離する.

```python
# pochitrain/training/evaluator.py
class Evaluator:
    """検証・混同行列計算を担当."""
    def validate(self, model, val_loader, criterion, device) -> Dict[str, float]
    def compute_confusion_matrix(self, predicted, targets, num_classes) -> Tensor
    def log_confusion_matrix(self, cm, epoch, workspace_path) -> None
```

移動元:
- `PochiTrainer.validate()`
- `PochiTrainer._compute_confusion_matrix_pytorch()`
- `PochiTrainer._log_confusion_matrix()`

`PochiPredictor.calculate_accuracy()` も汎用メソッドとしてここに集約する.

### Phase 2: TrainingConfigurator の抽出

`training/training_configurator.py` を新規作成し, optimizer/scheduler/criterion 構築ロジックを分離する.

```python
# pochitrain/training/training_configurator.py
class TrainingConfigurator:
    """optimizer, scheduler, criterion の構築を担当."""
    def configure(self, model, config) -> TrainingComponents
    # 層別LR関連もここに集約
    def _build_layer_wise_param_groups(...)
    def _get_layer_group(...)
    def _log_layer_wise_lr(...)
```

移動元:
- `PochiTrainer.setup_training()`
- `PochiTrainer._build_layer_wise_param_groups()`
- `PochiTrainer._get_layer_group()`
- `PochiTrainer._log_layer_wise_lr()`
- `PochiTrainer._get_base_learning_rate()`
- `PochiTrainer.is_layer_wise_lr_enabled()`
- `PochiTrainer.get_layer_wise_learning_rates()`

### Phase 3: CheckpointStore ラッパーの削除

`save_checkpoint()`, `save_best_model()`, `save_last_model()`, `load_checkpoint()` は
`CheckpointStore` への単純委譲のため削除し, `train()` 内で `self.checkpoint_store` を直接使う.

### Phase 4: ワークスペース操作の移動

**PochiTrainer 側:**
- `save_training_config()`, `save_dataset_paths()`, `save_image_list()`, `get_workspace_info()` を削除
- CLI (`pochi.py`) から `PochiWorkspaceManager` を直接使用する

**PochiPredictor 側:**
- `export_results_to_workspace()`, `save_confusion_matrix_image()`, `predict_with_paths()` を削除
- `inference/result_exporter.py` を新規作成し, 推論結果出力を統括する Facade として機能させる
- `_ensure_inference_workspace()`, `get_inference_workspace_info()` は `InferenceWorkspaceManager` 側に集約

```python
# pochitrain/inference/result_exporter.py
class InferenceResultExporter:
    """推論結果の出力を統括する Facade."""
    def export(self, predictions, confidences, image_paths, ...) -> Tuple[Path, Path]
    def save_confusion_matrix_image(...) -> Path
```

## リファクタリング後のフォルダ構造 (差分のみ)

```
pochitrain/
├── pochi_trainer.py              ← ~360行 (train + train_epoch のみ)
├── pochi_predictor.py            ← ~230行 (predict + get_model_info のみ)
│
├── training/
│   ├── checkpoint_store.py       ← 既存 (変更なし)
│   ├── metrics_tracker.py        ← 既存 (変更なし)
│   ├── early_stopping.py         ← 既存 (変更なし)
│   ├── evaluator.py              ← 【新規】validate, 混同行列, 精度計算
│   └── training_configurator.py  ← 【新規】optimizer/scheduler/criterion + 層別LR
│
├── inference/
│   ├── csv_exporter.py           ← 既存 (変更なし)
│   └── result_exporter.py        ← 【新規】推論結果出力の Facade
│
└── utils/
    └── directory_manager.py      ← 既存 (dataset_paths 保存等を集約)
```

## 対象ファイル

- `pochitrain/pochi_trainer.py` — 責務外メソッドの削除
- `pochitrain/pochi_predictor.py` — 責務外メソッドの削除
- `pochitrain/training/evaluator.py` — 新規作成
- `pochitrain/training/training_configurator.py` — 新規作成
- `pochitrain/inference/result_exporter.py` — 新規作成
- `pochitrain/cli/pochi.py` — 新クラスの呼び出しに変更
- `pochitrain/utils/directory_manager.py` — ワークスペース操作の集約
- `tests/` — 対応するテストの追加・修正

## Labels

`refactor`

## Branch

`refactor/slim-trainer-predictor`
